### PR TITLE
Finish porting loaders, printers and savers

### PR DESCRIPTION
**Pull request by raxyte**
Ports:
- `load_amqp` and `save_amqp`
- `load_ftp` and `save_ftp`
- `load_nics`
- `read_pcap`

Implements `write_ndjson`.

- Closes https://github.com/tenzir/issues/issues/2186

### CODE REVIEW

File: libtenzir/builtins/formats/json.cpp [hunk 1]
```diff
@@ -1321,6 +1321,23 @@ class write_json_plugin final : public virtual operator_plugin2<write_json> {
   }
 };
 
+class write_ndjson_plugin final : public virtual operator_plugin2<write_json> {
+public:
+  auto name() const -> std::string override {
+    return "write_ndjson";
```
Comment (dominiklohmann) [inline libtenzir/builtins/formats/json.cpp:1327]: With this, should we also remove the `ndjson=<bool>` option from `write_json`? That'd mirror the `read_json` and `read_ndjson` operators.

File: plugins/amqp/builtins/plugin.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          .emit(ctx);
+        return failure::promise();
+      }
+    }
+    if (options) {
+      // For all string keys, we don't attempt automatic conversion.
+      const auto strings = std::set<std::string>{
+        "hostname", "vhost", "sasl_method", "username", "password"};
+      for (auto& [k, v] : options->inner) {
+        if (auto str = caf::visit(stringify, v)) {
```
Comment (dominiklohmann) [inline plugins/amqp/builtins/plugin.cpp:107]: You probably need to rebase this PR, this line shouldn't compile anymore.

File: web/docs/tql2/operators/read_pcap.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+[pcapng-rfc]: https://www.ietf.org/archive/id/draft-tuexen-opsawg-pcapng-05.html
+
+:::note PCAPNG
+The current implementation does *not* support [PCAPNG][pcapng-rfc]. Please
+[reach out](/discord) if you would like to see support.
+:::
+
+The structured representation of packets has the `pcap.packet` schema:
+
+```yaml
```
Comment (IyeOnline) [inline web/docs/tql2/operators/read_pcap.md:25]: Change this to a table like https://docs.tenzir.com/tql2/operators/metrics

and change the type of `data` to `blob`.

File: web/docs/tql2/operators/load_nic.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+When setting `emit_file_headers` every chunk gets its own PCAP file header, as
+opposed to just the very first. This yields a continuous stream of concatenated
+PCAP files.
+
+The [`pcap`](read_pcap.md) parser can handle such concatenated traces, and
+optionally re-emit thes file headers as separate events.
+
+## Examples
+
+Read PCAP packets from `eth0`:
```
Comment (IyeOnline) [inline web/docs/tql2/operators/load_nic.md:46]: Make this an h3 instead

File: web/docs/tql2/operators/write_json.md [hunk 5]
```diff
@@ -3,17 +3,13 @@
 Transforms the input event stream to JSON byte stream.
 
 ```tql
-write_json [ndjson=bool, color=bool]
+write_json [color=bool]
 ```
 
 ## Description
 
 Transforms the input event stream to JSON byte stream.
```
[the remark this answers is not in the document]
Comment (IyeOnline) [reply web/docs/tql2/operators/write_json.md:12]: Note sure if we really need/want these, but I think they could be helpful.

**Review by IyeOnline — APPROVED** [review]
Giving this tentative approval now, assuming at least the AMQP config issue gets addressed.

**Pull request merged**